### DISCUSSION TITLE
Cannot use style='transition-delay:...' within timelineVariable

### DISCUSSION THREAD

**Discussion by angoodkind** [jspsych/jsPsych#1576] [Q&A]
I am using a modified jsPsych plugin called [jspsych-html-keyboard-multiple-response](https://github.com/CrumpLab/jspsychrexamples/blob/master/Stroop_multiple_response/experiment/jspsych-6/plugins/jspsych-html-keyboard-multiple-response.js), which is based on the `html-keyboard-response` plugin.

In my example below, there should be:
1) an image displayed immediately 
2) a 3 second delay, followed by a text prompt
3) the ability to enter free text below the text prompt

(1) and (3) work fine, but I cannot get the 3 second delay in (2) to work. Since the stimulus html is originally entered as a string, is there something different I should do? I've already tried adding a `<style>` block to the html file, and modifying jspsych.css with a new selector. Any idea where I'm going wrong?

```
<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<title>Keystroke Timing Demo</title>

<script src="jspsych-6/jspsych.js"></script>
<script src="jspsych-6/plugins/jspsych-html-keyboard-response.js"></script>
<script src="jspsych-6/plugins/jspsych-html-keyboard-multiple-response.js"></script>
<link rel="stylesheet" href="jspsych-6/css/jspsych.css"></link>

</head>

<body>

<script> var  test_stimuli =  [
  {
    "stimulus": "<img src='green-blue-bus.png' width='300' height='300'><p style='transition-delay:3s'>How would you describe the highlighted bus?<\/p>",
    "data": {
      "stim_type": "congruent"
    }
  }
] ; </script>

<script type="text/javascript">
/* experiment parameters */
var reps_per_trial_type = 1;

/* defining test timeline */
var test = {
  timeline: [{
    type: 'html-keyboard-multiple-response',
    response_ends_trial: false,
    stimulus: jsPsych.timelineVariable('stimulus'),
    data: jsPsych.timelineVariable('data'),
  }],
  timeline_variables: test_stimuli,
  sample: {type: 'fixed-repetitions', size: reps_per_trial_type}
};

/*set up experiment structure*/
var timeline = [];
timeline.push(test);

/*start experiment*/
jsPsych.init({
    timeline: timeline,
    on_finish: function() {
            jsPsych.data.displayData();
        }
});
</script>

</body>
</html>
```

**Comment by becky-gilbert** [comment]
Hi @angoodkind! I haven't used this property before, but based on the [transition docs](https://www.w3schools.com/css/css3_transitions.asp), it looks like the transition-delay property has to be used in conjunction with a transition property. I don't see a value for the CSS transition in your stimulus HTML. Is it set somewhere else?

**Reply by angoodkind** [reply]
Hi @becky-gilbert , thanks so much for getting back to me. I do not have a transition property set anywhere else (that I know of). What would that look like?

**Reply by becky-gilbert** [reply]
Hi @angoodkind, I'm wondering if the CSS transition property is the right way to go here. Admittedly I don't know much about it, but all the examples I see are related to transitioning a CSS style property when the element's class changes. In your case, I don't think the stimulus class is changing after 3 seconds, which makes it hard to set up a transition. Another option would be to use [CSS animations/keyframes](https://www.w3schools.com/css/css3_animations.asp) instead, which allows you to change the element's appearance over time, without having to change classes. If you do it this way, then I think you'll want to change the element's opacity.

If it were me, I would do this by initially setting the prompt's visibility to "hidden" in the stimulus HTML, and using the [`jsPsych.pluginAPI.setTimeout` function](https://www.jspsych.org/core_library/jspsych-pluginAPI/#jspsychpluginapisettimeout) to change its visibility to "visible" after 3 seconds. You could do this in the trial's [on_start](https://www.jspsych.org/overview/plugins/#the-on_start-parameter) or [on_load](https://www.jspsych.org/overview/plugins/#the-on_load-parameter) function. Here's an example (not tested...):

```js
var  test_stimuli =  [
  {
    "stimulus": "<img src='green-blue-bus.png' width='300' height='300'><p id='prompt' style='visibility:hidden;'>How would you describe the highlighted bus?</p>",
    "data": {
      "stim_type": "congruent"
    }
  }
] 

var test = {
  timeline: [{
    type: 'html-keyboard-multiple-response',
    response_ends_trial: false,
    stimulus: jsPsych.timelineVariable('stimulus'),
    data: jsPsych.timelineVariable('data'),
    on_load: function() {
      jsPsych.pluginAPI.setTimeout(function() {
        document.getElementById('prompt').style.visibility = 'visible';
      }, 3000);
    }
  }],
  timeline_variables: test_stimuli,
  sample: {type: 'fixed-repetitions', size: reps_per_trial_type}
};
```

A third option is to modify the plugin file. You could add new trial parameters for the `prompt` and `prompt_delay`, add the prompt to the trial's HTML string with its visibility set to "hidden", and then add a `jsPsych.pluginAPI.setTimeout` function as above, but where the `setTimeout` delay is `trial.prompt_delay`, rather than 3000.

I hope this helps. Let us know if you need further guidance.
